### PR TITLE
automated duplicating imported samples modifications

### PR DESCRIPTION
**Pull request by robGG1997**
this will automate the dev process of duplicating modifications to the imported package sample
uses a bash script to remove and duplicate samples folder
uses a git actions workflow duplicate_sample.yml to check for changes to imported package sample to run duplicate_package_sample.sh and commit and push any changes if they exist
This will only be automated to add a commit when a push happens to main (successful pull request)

### CODE REVIEW

**Comment by RyRy79261** [comment]
@robGG1997 We might want a shell script for Windows as well

**Comment by sneakzttv** [comment]
> @robGG1997 We might want a shell script for Windows as well

Just use git, it gives you bash for windows https://git-scm.com/downloads

**Review by sneakzttv — DISMISSED** [review]

**Comment by sneakzttv** [comment]
> @sneakzttv I normally use WSL2 so anything git related runs out of that section, but the issue is pathing buggers out for all the .Net stuff for me, so I just run directly out of windows

Fair

**Comment by sneakzttv** [comment]
hmm...added a comment, initialized the build then reimported sample via the package manager and the comment had been reset.

**Comment by robGG1997** [comment]
> hmm... forked the branch, added a comment and made a commit. Initialized the build then reimported sample via the package manager and the comment had been reset. If this is because the branch was forked then ignore me, otherwise it was working fine on initial import. @robGG1997 just confirm when you can and i'll approve based on said outcome.

Have you set up pre-commit in .git/hooks?

**Comment by robGG1997** [comment]
> @robGG1997 We might want a shell script for Windows as well

based on the new changes the bash script is going to be used by a GitHub action on a push to main, no need to use it manually

**Comment by robGG1997** [comment]
> @robGG1997 We might want a shell script for Windows as well

based on the new changes the bash script is going to be used by a GitHub action on a push to main, no need to use it manually

**Comment by sneakzttv** [comment]
so...we wait?

**Comment by robGG1997** [comment]
> so...we wait?

When @kalambet approves this we're good to go

**Comment by kalambet** [comment]
I'm still concern a bit with extra commit. There is no urgency in solving this so far, however I'll try to finalize my thoughts till the end of the day. Sorry :(

File: .github/workflows/duplicate_sample.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/checkout@v2
+
+      - name: Run duplicate script file
+        run: bash ${GITHUB_WORKSPACE}/duplicate_package_sample.sh
+
+      - name: Commit and Push
+        run: |
+          git config --global user.email "robel@chainsafe.io"
+          git config --global user.name "Robel Getnet"
+          git config --global user.username "robGG1997"
```
Comment (kalambet) [inline .github/workflows/duplicate_sample.yaml:24]:
```suggestion
          git config --global user.email "${{ github.actor }}@users.noreply.github.com"
          git config --global user.name "${{ github.actor }}"
```

Comment (kalambet) [reply .github/workflows/duplicate_sample.yaml:24]: I think there is no better solution right now. 
However let's personalize commit to the person who initiated workflow to run.

Comment (robGG1997) [reply .github/workflows/duplicate_sample.yaml]: done, now the commit author will be whoever initiated the workflow, I've also tested this

**Review by kalambet — APPROVED** [review]
LGTM

**Comment by kalambet** [comment]
@robGG1997 Please update the branch with the changes from `main`

**Review by sneakzttv — APPROVED** [review]

**Review by kantagara — APPROVED** [review]

**Pull request merged**